### PR TITLE
Add postgres support

### PR DESCRIPTION
**Pull request by damdo**
Add support for using postgres as a database for GUS.

### CODE REVIEW

File: internal/gusserver/gusserver.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ import (
 	// It’s twice as slow compared to using github.com/mattn/go-sqlite3 (SQLite
 	// with cgo), but that’s still good enough for what we need:
 	// https://datastation.multiprocess.io/blog/2022-05-12-sqlite-in-go-with-and-without-cgo.html
+	_ "github.com/lib/pq"
```
Comment (stapelberg) [inline internal/gusserver/gusserver.go]: Can you move this to a separate block (separated by blank line) please?

File: internal/gusserver/gusserver.go [hunk 2]
```diff
@@ -109,13 +110,14 @@ func (s *server) heartbeat(w http.ResponseWriter, r *http.Request) error {
 	return nil
 }
 
-func newServer(databaseDir string) (*server, *http.ServeMux, error) {
-	db, err := sql.Open("sqlite", databaseDir+"?mode=rwc")
+func newServer(databaseType, databaseSource string) (*server, *http.ServeMux, error) {
+	fmt.Println(databaseType, databaseSource)
```
Comment (stapelberg) [inline internal/gusserver/gusserver.go]: Remove debugging left-over (or turn into a proper log.Printf() message)

File: internal/gusserver/gusserver.go [hunk 3]
```diff
@@ -133,15 +135,17 @@ func newServer(databaseDir string) (*server, *http.ServeMux, error) {
...
-		databaseDir = flag.String("database_dir", "/var/lib/gus", "database directory for GUS internal state. the special value :memory: stores state in memory")
+		listen         = flag.String("listen", "localhost:8655", "[host]:port listen address")
+		databaseType   = flag.String("database_type", "sqlite", "can be one of: sqlite, postgres")
+		databaseSource = flag.String("database_source", ":memory:", "database source for GUS internal state. can be :memory: (default. stores state in memory), directory path (sqlite) or an connection DSN (postgres. reference: https://pkg.go.dev/github.com/lib/pq#hdr-Connection_String_Parameters)")
 	)
 	flag.Parse()
 
-	if *databaseDir != ":memory:" {
-		*databaseDir = filepath.Join(*databaseDir, "gus.db")
+	if *databaseType == "sqlite" {
```
Comment (stapelberg) [inline internal/gusserver/gusserver.go]: I think you need both conditionals here: *databaseType == "sqlite" && *databaseDir != ":memory:"

**Review by stapelberg — CHANGES_REQUESTED** [review]

File: internal/gusserver/gusserver.go (hunk 2 above)
Comment (damdo) [reply internal/gusserver/gusserver.go]: 👍  turned it into a `log` line

**Review by stapelberg — APPROVED** [review]

**Pull request merged**